python/tests/test_dataset.py

```diff
@@ -56,54 +56,127 @@ def test_del_item_missing():
         del d['not an item']
 
 
-def test_set_coord():
-    d = sc.Dataset()
-    d.set_coord(Dim.X, sc.Variable(1.0))
-    assert len(d) == 0
+def test_coord_setitem():
+    var = sc.Variable([Dim.X], values=np.arange(4))
+    d = sc.Dataset({'a': var}, coords={Dim.X: var})
+    with pytest.raises(RuntimeError):
+        d[Dim.X, 2:3].coords[Dim.Y] = sc.Variable(1.0)
+    with pytest.raises(RuntimeError):
+        d['a'].coords[Dim.Y] = sc.Variable(1.0)
+    d.coords[Dim.Y] = sc.Variable(1.0)
```

Comment:
:+1: 
I also checked that changing the values of the current coordinates
```python
d.coords[Dim.X] = sc.Variable([Dim.X], values=np.arange(4, 8))
```
is fine, but changing to a new array with a different shape
```python
d.coords[Dim.X] = sc.Variable([Dim.X], values=np.arange(4, 6))
```
fails.